libraries/ts-command-line/src/CommandLineDefinition.ts

```diff
@@ -21,9 +21,20 @@ export interface IBaseCommandLineDefinition {
    * Documentation for the flag, that will be shown when invoking the tool with "--help"
    */
   description: string;
+
+  /**
+   * If true and no "getDefaultValue" function is provided, the parameter must be provided.
+   */
+  required?: boolean;
+
+  /**
+   * A default value for the parameter if no value was provided. If this property is undefined and the "required" flag
+   *  is set, validation will fail.
```

Comment:
This doesn't make sense.  How can something be "required" and also have a default value?

Reply:
That is a good question. @iclanton , please sort out this before merging.